control-plane/agents/core/src/pool/specs.rs

```diff
@@ -382,37 +382,43 @@ impl ResourceSpecsLocked {
                     return Err(SvcError::ReplicaNotFound {
                         replica_id: request.uuid.clone(),
                     });
-                } else if spec.share == status.share && spec.share != Protocol::Off {
-                    return Err(SvcError::AlreadyShared {
-                        kind: ResourceKind::Replica,
-                        id: request.uuid.to_string(),
-                        share: spec.share.to_string(),
-                    });
+                } else {
+                    // validate the operation itself against the spec and status
+                    if spec.share != status.share {
+                        return Err(SvcError::NotReady {
+                            kind: ResourceKind::Pool,
+                            id: request.uuid.to_string(),
+                        });
+                    } else if request.protocol == spec.share || spec.share.shared() {
```

Comment:
Again, can we just check `spec.share.shared()`?

Reply:
Yep same here